snuba/datasets/processors/spans_processor.py

```diff
@@ -46,14 +46,15 @@ class SpansMessageProcessor(DatasetMessageProcessor):
     from the transactions topic and de-normalize it into the spans table.
     """
 
-    def __extract_timestamp(self, timestamp_sec: float) -> Tuple[datetime, int]:
+    def __extract_timestamp(self, timestamp_ms: int) -> Tuple[int, int]:
         # We are purposely using a naive datetime here to work with the rest of the codebase.
         # We can be confident that clients are only sending UTC dates.
-        timestamp = _ensure_valid_date(datetime.utcfromtimestamp(timestamp_sec))
+        timestamp = _ensure_valid_date(datetime.utcfromtimestamp(timestamp_ms / 1000))
         if timestamp is None:
             timestamp = datetime.utcnow()
         milliseconds = int(timestamp.microsecond / 1000)
-        return timestamp, milliseconds
+        unix_timestamp = int(timestamp.replace(tzinfo=timezone.utc).timestamp())
```

Comment:
I have to replace the timezone as our dev environments are not on UTC time usually but it's not something a production system need.

https://docs.python.org/3/library/datetime.html#datetime.datetime.timestamp

Reply:
I'd skip the conversion from int to datetime and back again, then you won't have to deal with timezone at all.

Something like:
```
is_valid = _ensure_valid_date(datetime.utcfromtimestamp(timestamp_ms / 1000)) is not None
timestamp_ms = timestamp_ms if is_valid else int(time.time() * 1000)
return int(timestamp_ms / 1000), int(timestamp_ms % 1000)
```